addons/sale/models/sale_order.py

```diff
@@ -1927,10 +1927,12 @@ def _get_product_documents(self):
         return self._filter_product_documents(documents).sorted()
 
     def _filter_product_documents(self, documents):
+        customer_lang = self.partner_id.lang or self.partner_id.parent_id.lang
+
         return documents.filtered(
             lambda document:
-                document.attached_on_sale == 'quotation'
-                or (self.state == 'sale' and document.attached_on_sale == 'sale_order')
+                document.attached_on_sale in ('quotation', 'sale_order') and
+                document.lang in (False, customer_lang)
```

Comment:
```suggestion
                 document.lang in (False, customer_lang) and
                document.attached_on_sale == 'quotation'
                or (self.state == 'sale' and document.attached_on_sale == 'sale_order')
```
To make condition same as before